### PR TITLE
add open hub to state

### PR DESCRIPTION
**Pull request by yonip23**

### CODE REVIEW

File: src/binary/pollDownloadProgress.ts [hunk 1]
```diff
@@ -9,14 +8,15 @@ import {
   DownloadStatus,
   State,
 } from "./state";
+import getStateA from "./requests/state";
```
Comment (yonip23) [inline src/binary/pollDownloadProgress.ts:11]: change this to a meaningful name (its wip)

File: src/binary/requests/state.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { StateType } from "../../consts";
+import { State } from "../state";
+import { getState } from "./requests";
+import { openConfigWithSource } from "../../commandsHandler";
+
+export default async function getStateA(): Promise<State | null | undefined> {
```
Comment (yonip23) [inline src/binary/requests/state.ts:6]: change this to a meaningful name (its wip)

Comment (boaz-codota) [reply src/binary/requests/state.ts:6]: you said so....

File: src/binary/requests/state.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  const state = await getState();
+
+  if (state?.should_open_hub) {
+    void openConfigWithSource(StateType.INFO)();
```
Comment (boaz-codota) [inline src/binary/requests/state.ts:10]: Is this firing a command? Looks weird

**Review by boaz-codota — APPROVED** [review]
Give better name

File: src/binary/requests/state.ts (hunk 3 above)
Comment (yonip23) [reply src/binary/requests/state.ts:10]: it opens the hub... im not sure what you mean

Comment (dimacodota) [reply src/binary/requests/state.ts:10]: why is it here?

Comment (yonip23) [reply src/binary/requests/state.ts:10]: where else should it be?

**Comment by yonip23** [comment]
closing as im going for a different implementation

**Comment by yonip23** [comment]
closing as im going for a different implementation

**Pull request closed without merging**